presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1570,6 +1571,7 @@ private Scope computeAndAssignOrderByScope(OrderBy node, Scope sourceScope, Scop
             // ORDER BY should "see" both output and FROM fields during initial analysis and non-aggregation query planning
             Scope orderByScope = Scope.builder()
                     .withParent(sourceScope)
+                    .withScopeId(outputScope.getScopeId())
```

Comment:
> Scopes are not meant to uniquely identify resolved fields. They are just field (names) resolution mechanism that models hierarchy between relations.

This is not affected by my change. Remember: for identification purposes, only the _effectively resolving_ scope matters.

> Why not just move RelationId out of RelationType into Scope?

It would work, but I'd like to hear others' opinion too, before diving in (ping @kokosing  and @martint if you find time)
I think we want to identify "source of field" or "source relation" so yeah, `RelationId` as the name makes more sense.

> ORDER BY sees both output and from fields but top QuerySpecification sees only output fields of lower relation

Yes, but we have options how to model that.
- as is now: create scope with copy of output fields
- enhance scopes to support two parents and don't copy anything

This might be viewed as orthogonal to the main plot, but interesting nonetheless.